src/rpk_entity.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2015-2021 Brian Smith.
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHORS DISCLAIM ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR
+// ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+// ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+// OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+use crate::der;
+use crate::error::Error;
+use crate::signed_data;
+use pki_types::{CertificateDer, SignatureVerificationAlgorithm};
+/// An RPK certificate, used for RawPublicKeys connections as specified in RFC7250.
+/// This certificate can be used for the Raw Public Keys sent from client to server and from server to client
+///
+/// Contrary to `EndEntityCert`, this struct only contains the Subject Public Key Info.
+///
+/// * `RPKCert.verify_signature`: Verifies that the signature of server's
+///   `ServerKeyExchange` message is valid for the server's certificate.
+///
+/// Client certificate processing in a TLS connection consists of analogous
+/// steps. All of these steps are necessary:
+///
+/// * `RPK.verify_signature`: Verify that the client's signature in
+///   its `CertificateVerify` message is valid using the public key from the
+///   client's certificate.
+pub struct RPKCert<'a> {
+    spki: untrusted::Input<'a>,
+}
+
+impl<'a> TryFrom<&'a CertificateDer<'a>> for RPKCert<'a> {
+    type Error = Error;
+
+    /// Parse the ASN.1 DER-encoded SPKI encoding of the certificate
+    /// `cert_der`.
+    fn try_from(cert: &'a CertificateDer<'a>) -> Result<Self, Self::Error> {
+        Ok(Self {
+            spki: untrusted::Input::from(cert)
+                .read_all(Error::BadDer, |input| {
+                    der::expect_tag(input, der::Tag::Sequence)
+                })
+                .unwrap(),
+        })
```

Comment:
Leg uit dat dit de sequence wrapper weghaalt zodat het aligned is met de spki-format die nodig is voor de onderliggende verify signature function